src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -1405,12 +1409,36 @@ void AutomationInstrumentClipView::editPadAction(bool state, uint8_t yDisplay, u
 					    getParameterKnobPos(modelStackWithParam, getPosFromSquare(leftPadSelectedX)) + kKnobPosOffset;
 					indicator_leds::setKnobIndicatorLevel(0, knobPos);
 
-					knobPos =
-					    getParameterKnobPos(modelStackWithParam, getPosFromSquare(rightPadSelectedX)) + kKnobPosOffset;
+					int32_t effectiveLength;
+
+					if (instrument->type == InstrumentType::KIT && !instrumentClipView.getAffectEntire()) {
+						ModelStackWithNoteRow* modelStackWithNoteRow = clip->getNoteRowForSelectedDrum(modelStack);
+
+						effectiveLength = modelStackWithNoteRow->getLoopLength();
+					}
+					else {
+						//this will differ for a kit when in note row mode
+						effectiveLength = clip->loopLength;
+					}
+
+					int32_t squareRightEdge = getPosFromSquare(rightPadSelectedX + 1);
+					uint32_t squareStart = std::min(effectiveLength, squareRightEdge) - kParamNodeWidth;
+					knobPos = getParameterKnobPos(modelStackWithParam, squareStart) + kKnobPosOffset;
 					indicator_leds::setKnobIndicatorLevel(1, knobPos);
 
```

Comment:
This sets the mod position to grab the value at the position of the second pad pressed (last pad pressed). This is needed so that when you use the audition pads, it grabs the last param value set.